syncstorage/storage/memcached.py

```diff
@@ -944,23 +945,31 @@ def del_item(self, userid, item):
             raise ItemNotFoundError
         return modified
 
-    def get_cached_batches(self, userid):
-        return self.cache.gets(self.get_batches_key(userid))
+    def get_cached_batches(self, userid, ts=None):
+        if ts is None:
+            ts = get_timestamp()
+        ts = int(ts)
+        bdata, bcasid = self.cache.gets(self.get_batches_key(userid))
+        # Remove any expired batches, but let the
+        # calling code write it back out to memcache.
+        if bdata:
+            for batchid, batch in bdata.items():
+                if batch["created"] + BATCH_LIFETIME < ts:
+                    del bdata[batchid]
+        return bdata, bcasid
```

Comment:
This consolidates the logic for expiring batches from memcache in a single location.